### PR TITLE
MueLu: Adding experimental node comm support

### PR DESCRIPTION
**Pull request by csiefer2**
Auto-PR for SHA 009a8bf

### CODE REVIEW

**Comment by csiefer2** [comment]
Checked on serrano w/ the debugging stuff on and it does what we want.

File: packages/muelu/src/Utils/MueLu_Utilities.cpp [hunk 1]
```diff
@@ -248,4 +256,43 @@ bool IsParamValidVariable(const std::string& name)
...
+       int myaddr = (int) htonl(inet_network(inet_ntoa(*(struct in_addr *)host->h_addr)));
+       
+       // All-to-all exchange of address integers
+       std::vector<int> addressList(numRanks);
+       Teuchos::gatherAll(*baseComm,1,&myaddr,numRanks,&addressList[0]);
+
+       // Sort!
+       std::sort(addressList.begin(),addressList.end());
+
+       // Find the number of node-buddies I have
```
Comment (jhux2) [inline packages/muelu/src/Utils/MueLu_Utilities.cpp]: This comment misled me.  The code below is identifying the node ID of the current MPI rank, counting up from zero, isn't it?  All MPI ranks on a node should calculate the same node ID.

Comment (csiefer2) [reply packages/muelu/src/Utils/MueLu_Utilities.cpp]: Oh, yeah, that comment is not correct.

**Comment by csiefer2** [comment]
> Wait, woah, I thought several MPI implementations (OpenMPI and MPICH at least) came with built-in "node communicators" that are just named constants you can use. Wouldn't that be `MPI_COMM_NETWORK`? Maybe that was a thing that went away, but I see some Zoltan code that refers to `MPI_COMM_NODE`, `MPI_COMM_SOCKET`, and `MPI_COMM_CACHE`.

I think those are version specific.  I do not believe there's a portable way of getting those from MPI directly.

**Pull request merged**